framework/auth/views.py

```diff
@@ -251,6 +288,68 @@ def confirm_email_get(token, auth=None, **kwargs):
     ))
 
 
+@collect_auth
+def unconfirmed_email_remove(auth=None):
+    """Called at login if user cancels their merge or email add.
+    methods: DELETE
+    """
+    user = auth.user
+    if user is None:
+        raise HTTPError(http.NOT_FOUND)
```

Comment:
That said, I just discovered the behavior for `must_be_logged_in` is incorrect for JSON requests. I've filed a ticket (https://openscience.atlassian.net/browse/OSF-6317). I still think it's still ok to use that decorator for now though.